lib/rules/require-padding-newlines-before-line-comments.js

```diff
@@ -87,6 +88,13 @@ module.exports.prototype = {
                 return;
             }
 
+            var firstToken = file.getFirstTokenOnLine(comment.loc.start.line);
+
+            // Should not consider code and comment on the same line (#1194)
+            if (firstToken !== undefined && firstToken.type !== 'EOF') {
```

Comment:
Why do we need to check for undefined and EOF? It seems like we could just check if the first token on the line **IS** the comment. If the comment isn't the first token on the line, return out.